prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+
+import csv
+import json
+import os.path
+import zipfile
+from collections import defaultdict
+
+import networkx as nx
+import pandas as pd
+from haversine import Unit, haversine
+
+Max_Value = 3000
+Min_Value = 0
+coord_precision = ".9f"
+default_base_kv = -999999.0
+
+
+def get_Zone(Z_csv):
+    """Generate a dictionary of zone using the zone.csv
+
+    :param str Z_csv: path of the zone.csv file
+    :return: (*dict*) -- a dict mapping the STATE to its ID.
+    """
+
+    zone = pd.read_csv(Z_csv)
+
+    # Create dictionary to store the mapping of states and codes
+    zone_dic = {}
+    for i in range(len(zone)):
+        zone_dic[zone["STATE"][i]] = zone["ID"][i]
+    return zone_dic
+
+
+def Clean(E_csv, zone_dic):
+    """Clean data; remove substations which are outside the United States or not available.
+
+    :param str E_csv: path of the HIFLD substation csv file
+    :param dict zone_dic: zone dict as returned by :func:`get_Zone`
+    :return: (*pandas.DataFrame*) -- a pandas Dataframe storing the substations after dropping the invalid ones.
+    """
+
+    csv_data = pd.read_csv(E_csv)
+    Num_sub = len(csv_data)
+    row_indexs = []
+    for i in range(Num_sub):
+        if (
+            (csv_data["STATE"][i] not in zone_dic)
+            or (csv_data["STATUS"][i] != "IN SERVICE")
+            or (csv_data["LINES"][i] == 0)
+        ):
+            row_indexs.append(i)
+    clean_data = csv_data.drop(labels=row_indexs)
```

Comment:
Is this equivalent to:
```python
clean_data = csv_data.query(
    "STATE in @zone_dic and STATUS == 'IN SERVICE' and LINES != 0"
)
```
?

Reply:
Good catch, this is way more concise! will modify. thanks.